cirq/qis/measures.py

```diff
@@ -132,10 +133,7 @@ def fidelity(
 
     # Use QuantumState machinery for the general case
     if qid_shape is None:
-        # coverage: ignore
-        raise NotImplementedError(
-            'Qid shape inference not yet implemented. Please specify the qid shape explicitly.'
-        )
+        qid_shape = infer_qid_shape(state1, state2)
```

Comment:
I think a message to specify the qid shape explicitly is still useful here to help the user. Otherwise only the inference errors will be shown which are a bit more obscure.

Reply:
Done.